pkg/pillar/worker/opt.go

```diff
@@ -0,0 +1,51 @@
+package worker
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/lf-edge/eve/pkg/pillar/base"
+)
+
+const (
+	// DefaultQueue default queue depth
+	DefaultQueue = 5
+	// DefaultParallel default parallelism of just 1 worker; no more supported yet
+	DefaultParallel = 1
```

Comment:
My thought a while back was that it is easier to add the parallelism on top of the single worker, in particular if we want it to dynamically create gorotines based on load (up to some max).

To avoid head of line blocking the submission (in the channel) needs to be non-blocking so that the pool code can try the different workers and if all fail create a new one (up to some max).

So now clear we want this parallelism inside the worker but in a package on top of it.

Reply:
I am not adding any parallelism right now, just simplifying the interface and removing as much code duplication as possible while there are only two consumers (volumemgr and baseosmgr). 

My assumption is that at some point we will add parallelism. How that will work - parallel per Kind or parallel across all - is TBD.

Reply:
My point is that parallelism might be better done in a separate package (called workerpool?) which uses the worker package. By adding this config you are already committing to putting it in this worker package.

Reply:
That is a very good point. I intended to put it in this package, but having a `NewPool` option, but hadn't gotten to it yet. It needs more work than I could do during this last rush.

I can remove this for now to keep the option open both ways.